protostar/cheatable_starknet/cheatables/cheatable_cached_state.py

```diff
@@ -48,6 +55,37 @@ def __init__(
         self.contract_address_to_block_timestamp: dict[Address, int] = {}
         self.contract_address_to_block_number: dict[Address, int] = {}
 
+    def add_mocked_response(
+        self,
+        target_address: Address,
+        entrypoint: Selector,
+        mocked_response: CairoData,
+    ) -> int:
+        if target_address not in self.address_to_entrypoint_to_mocked_response:
+            self.address_to_entrypoint_to_mocked_response[target_address] = {}
+        self.address_to_entrypoint_to_mocked_response[target_address][
+            entrypoint
+        ] = mocked_response
+        mock_id = hash(str(target_address) + str(entrypoint))
+        self.mocked_response_id_to_address_and_selector[mock_id] = (
+            target_address,
+            entrypoint,
+        )
+        return hash(str(target_address) + str(entrypoint))
+
+    def get_mocked_response(
+        self, target_address: Address, entrypoint: Selector
+    ) -> Optional[CairoData]:
+        if target_address in self.address_to_entrypoint_to_mocked_response:
+            if (
+                entrypoint
+                in self.address_to_entrypoint_to_mocked_response[target_address]
+            ):
+                return self.address_to_entrypoint_to_mocked_response[target_address][
+                    entrypoint
+                ]
+        return None
```

Comment:
```suggestion
mocked_responses = self.address_to_entrypoint_to_mocked_response.get(target_address)
    if not mocked_responses:
        return None
    return self.address_to_entrypoint_to_mocked_response[target_address].get(entrypoint)
```

Reply:
`if not mocked_responses:` introduces a bug.  `mock_call(..., [])` won't have an effect.

Reply:
so please do `if mocked_responses is not None` instead

Reply:
But actually i think it won't, since the list will contain another list (empty), which is not falsy.

Reply:
You right. I thought `mocked_responses` is a `list` not a `dict`.

Reply:
You can change the name - i leave it up to you, wanted to show a simpler structure

Reply:
reduced to oneline
[simplify get_mocked_response](https://github.com/software-mansion/protostar/pull/1460/commits/d4672a0e9e103046ed0776a956e69a34dbea6a45)